src/main/java/org/hobbit/core/components/AbstractCommandReceivingComponent.java

```diff
@@ -301,7 +489,38 @@ private void initResponseQueue() throws IOException {
             responseQueueName = cmdChannel.queueDeclare().getQueue();
         }
         if (responseConsumer == null) {
-            responseConsumer = new QueueingConsumer(cmdChannel);
+            responseConsumer = new DefaultConsumer(cmdChannel) {
+                @Override
+                public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProperties properties,
+                        byte[] body) throws IOException {
+                    String key = properties.getCorrelationId();
+
+                    synchronized (responseFutures) {
+                        SettableFuture<String> future = null;
+                        if (key != null) {
+                            future = responseFutures.remove(key);
+                            if (future == null) {
+                                LOGGER.error("Received a message with correlationId ({}) not in map ({})", key, responseFutures.keySet());
+                            }
+                        } else {
+                            Iterator<SettableFuture<String>> iter = responseFutures.values().iterator();
+                            if (iter.hasNext()) {
+                                LOGGER.error("Received a message with null correlationId. This is an error unless the other component uses an older version of HOBBIT core library. Correlating with the eldest request as a workaround...");
+                                future = iter.next();
+                                iter.remove();
+                            } else {
+                                LOGGER.error("Received a message with null correlationId. This is an error unless the other component uses an older version of HOBBIT core library. There are no pending requests.");
```

Comment:
There are no pending requests so it is always an error. The middle of the message can be removed :thinking: